File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/Product/BenefitCoverMemberType.cs

using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;

namespace RMA.Service.ClientCare.Contracts.Entities.Product
{
    public class BenefitCoverMemberType
    {
        public int BenefitCoverMemberTypeId { get; set; } // BenefitCoverMemberTypeId (Primary key)
        public DateTime StartDate { get; set; } // StartDate
        public DateTime? EndDate { get; set; } // EndDate
        public int BenefitId { get; set; } // BenefitId
        public CoverMemberTypeEnum CoverMemberType { get; set; } // CoverMemberTypeId
        public bool IsDeleted { get; set; } // IsDeleted
        public string CreatedBy { get; set; } // CreatedBy (length: 50)
        public DateTime CreatedDate { get; set; } // CreatedDate
        public string ModifiedBy { get; set; } // ModifiedBy (length: 50)
        public DateTime ModifiedDate { get; set; } // ModifiedDate      
    }
}

File: Web/RMA-Web-Shell/projects/clientcare/src/app/product-manager/models/benefit-cover-member-type.ts
import { CoverMemberTypeEnum } from "projects/shared-models-lib/src/lib/enums/cover-member-type-enum";

export class BenefitCoverMemberType {

  benefitCoverMemberTypeId: number;
  startDate: Date;
  endDate: Date;
  coverMemberType: CoverMemberTypeEnum;
  benefitId: number;

  isDeleted: boolean;
  createdBy: string;
  modifiedBy: string;
  createdDate: Date;
  modifiedDate: Date;
}